curriculum/challenges/english/25-front-end-development/lab-confidential-email-page/66bba6fff611169359d9d36a.md

```diff
@@ -0,0 +1,249 @@
+---
+id: 66bba6fff611169359d9d36a
+title: Build a Confidential Email Page
+challengeType: 14
+dashedName: build-a-confidential-email-page
+demoType: onClick
+---
+
+# --description--
+
+**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab.
+
+**User Stories:**
+
+1. You should have a `main` element with an `id` of `email`.
+1. Your `#email` element should have `padding` of `50px`, a `top-margin` of `50px`, a `width` of `500px`, and a `border` that's `2px`.
+1. The total width of your `#email` element should be `500px`.
+1. You should have two `div` elements, one with an id of `confidential` and the other with an id of `top-secret`, within your `#email` element.
+1. Your `#confidential` and `#top-secret` elements should have a `display` of `inline-block`.
+1. Your `#confidential` and `#top-secret` elements should have a `padding`, a `left-margin`, and a `border`.
+1. Your `#confidential` and `#top-secret` elements should be rotated using a CSS transform.
+1. You should have at least three paragraph elements within your `#email` element.
+1. You should have a least three `span` elements with a class of `blurred`, within your paragraph elements.
+1. You should have a `blurred` selector that blurs the element `3px` using a CSS filter.
+
+# --hints--
+
+You should have a `main` element with an `id` of `email`.
+
+```js
+assert.exists(document.querySelector('main#email'));
+```
+
+You should have an `#email` selector that sets its elements' `padding` on all sides to `50px`.
+
+```js
+assert.equal(new __helpers.CSSHelp(document).getStyle('#email')?.padding, '50px');
+```
+
+You should have an `#email` selector that sets its elements' `margin-top` to `50px`.
+
+```js
+assert.equal(new __helpers.CSSHelp(document).getStyle('#email')?.marginTop, '50px');
+```
+
+You should have an `#email` selector that sets its elements' `width` to `500px`.
+
+```js
+assert.equal(new __helpers.CSSHelp(document).getStyle('#email')?.width, '500px');
+```
+
+Your `#email` element should have a `2px` border.
+
+```js
+assert.equal(new __helpers.CSSHelp(document).getStyle('#email')?.getPropertyValue('border-width'), '2px');
+```
+
+Your `#email` element should have a total width of `500px`.
```

Comment:
Maybe I am missing something. Is this different from the previous test "You should have an `#email` selector that sets its elements' `width` to `500px`." ?

Reply:
The idea here was to get them to add `box-sizing: border-box` so the element + padding and margin should be 500px.

That was a little tricky to test, so I adjusted the user story a little and changed the test to specifically check that.